### ISSUE TITLE
Migrating from 15.1.0 to 15.8.7 migrations fail | Cannot find module '@angular-devkit/core'

### ISSUE THREAD

**Issue by ak274** [nrwl/nx#15747]
### Current Behavior

I am trying to update nx project from 15.1.0 to the latest.. But after running the migrations I'm facing some issue in the screenshot below.
I have tried the workaround mentioned in one of the same issue here but no luck yet..
https://github.com/nrwl/nx/issues/7807#issuecomment-975448948

![image](https://user-images.githubusercontent.com/16129430/226091014-e1357522-8977-402a-a3aa-944d3c8b68e4.png)

### Expected Behavior

It should successfully update the workspace with latest version

### GitHub Repo
https://github.com/ak274/migration-issue

There are two branches 
main: with bare minimum workspace v15.1.0 with one angular app
after-migration: One is with after running the `nx migrate latest`

### Steps to Reproduce

1. 
Try to migrate a Nx workspace from 15.1.0 to 15.8.7.

### Nx Report

```shell
Node : 18.15.0
   OS   : win32 x64
   npm  : 9.5.0
   
   nx                      : 15.8.7
   @nrwl/js                : 15.8.7
   @nrwl/jest              : 15.8.7
   @nrwl/linter            : 15.8.7
   @nrwl/workspace         : 15.8.7
   @nrwl/angular           : 15.8.7
   @nrwl/cli               : 15.8.7
   @nrwl/cypress           : 15.8.7
   @nrwl/devkit            : 15.8.7
   @nrwl/eslint-plugin-nx  : 15.8.7
   @nrwl/nx-plugin         : 15.8.7
   @nrwl/tao               : 15.8.7
   @nrwl/webpack           : 15.8.7
   typescript              : 4.9.5
   ---------------------------------------
   Community plugins:
   @ng-bootstrap/ng-bootstrap : 14.0.1
   @ngrx/effects              : 15.3.0
   @ngrx/entity               : 15.3.0
   @ngrx/router-store         : 15.3.0
   @ngrx/store                : 15.3.0
   @koliveira15/nx-sonarqube  : 2.1.2
   @ngrx/schematics           : 15.3.0
   @ngrx/store-devtools       : 15.3.0
   ---------------------------------------
   Local workspace plugins:
         @luna/plugins
```

**Comment by leosvelperez** [comment]
Thanks for reporting this!

This issue has already been fixed in https://github.com/nrwl/nx/pull/15458 though the change has not been released yet. I'll try to get a patch version released with it. Please keep an eye on the releases to get the fix.

In the meantime, if you want to unblock yourself, locate the following migration in the `migrations.json` file and move it up above the other migrations for the `@nrwl/angular` package. Don't move it above other packages' migrations listed above the `@nrwl/angular` one, just put it as the first migration for `@nrwl/angular`.

```json
{
  "cli": "nx",
  "version": "15.7.0-beta.1",
  "description": "Install the required angular-devkit packages as we do not directly depend on them anymore",
  "factory": "./src/migrations/update-15-7-0/install-required-packages",
  "package": "@nrwl/angular",
  "name": "install-required-packages"
}
```

**Comment by ak274** [comment]
Thanks for looking into it. Looking forward to the next release.

But, I did tried the workaround as you explained above to move the install angular-devkit as first migration for package @nrwl/angular and I still faced the same issue. 

![image](https://user-images.githubusercontent.com/16129430/226341601-84cefc19-fc7c-4ded-9b36-82dbcb164b00.png)
![image](https://user-images.githubusercontent.com/16129430/226341781-127722c0-7653-430b-84fe-3c3cbc08f43b.png)

**Comment by ak274** [comment]
@leosvelperez I tried to migrate from `v15.1.0 to @nrwl/workspace@next`  which has the fix related to the issue though it generated the migrations in desired order by putting Install @angular/devkit as a first migration for @nrwl/angular but still the run migration fails with same error.

![image](https://user-images.githubusercontent.com/16129430/226837156-2ae0eac8-499f-438b-90ae-dd728c92e3a2.png)

**Comment by FrancescoBorzi** [comment]
hello, I have the same problem when trying to migrate from version `15.6.3` to `15.9.2`.

> NX Cannot find module '@angular-devkit/core'

-------

> This issue has already been fixed in https://github.com/nrwl/nx/pull/15458 though the change has not been released yet.

@leosvelperez is the fix supposed to be available in `15.9.2`? Am I doing something wrong?
